aptos-move/framework/aptos-framework/sources/delegation_pool.move

```diff
@@ -715,6 +722,119 @@ module aptos_framework::delegation_pool {
+
+        // convert existing commission to a 2-decimals precision percentage
+        operator_commission_percentage = operator_commission_percentage * 100;
+
+        let stake_pool_signer = account::create_signer_with_capability(&stake_pool_signer_cap);
+        coin::register<AptosCoin>(&stake_pool_signer);
+
+        // stake_pool_signer becomes owner of the stake pool and stores its `stake::OwnerCapability`
+        stake::deposit_owner_cap(&stake_pool_signer, owner_cap);
+
+        let inactive_shares = table::new<ObservedLockupCycle, pool_u64::Pool>();
+        table::add(
+            &mut inactive_shares,
+            olc_with_index(0),
+            pool_u64::create_with_scaling_factor(SHARES_SCALING_FACTOR)
+        );
+
+        let pool = DelegationPool {
+            active_shares: pool_u64::create_with_scaling_factor(SHARES_SCALING_FACTOR),
+            observed_lockup_cycle: olc_with_index(0),
+            inactive_shares,
+            pending_withdrawals: table::new<address, ObservedLockupCycle>(),
+            stake_pool_signer_cap,
+            // `staking_contract::distribute_internal` has already withdrawn any existing inactive stake
+            total_coins_inactive: 0,
+            operator_commission_percentage,
+            add_stake_events: account::new_event_handle<AddStakeEvent>(&stake_pool_signer),
+            reactivate_stake_events: account::new_event_handle<ReactivateStakeEvent>(&stake_pool_signer),
+            unlock_stake_events: account::new_event_handle<UnlockStakeEvent>(&stake_pool_signer),
+            withdraw_stake_events: account::new_event_handle<WithdrawStakeEvent>(&stake_pool_signer),
+            distribute_commission_events: account::new_event_handle<DistributeCommissionEvent>(&stake_pool_signer),
+        };
+
+        // Grant ownership of the existing stake to the participants of the destroyed staking contract.
+        // 1. `synchronize_delegation_pool` MUST NOT be called during this process because stake would be distributed as
+        // rewards instead of being individually allocated
+        // 2. partial governance voting remains disabled for simplicity
+
+        // allocate active stake to staker, any active commission has already been unlocked + allocated to operator
+        // no need to assert on minimum required stake as any next operation of delegator enforces it
+        buy_in_active_shares(&mut pool, staker_address, principal);
+
+        // allocate pending_inactive stake to staker, current operator, any previous operators still pending_inactive
+        while (pool_u64_iterable::shareholders_count(&distribution_pool) > 0) {
+            let recipients = pool_u64_iterable::shareholders(&distribution_pool);
+            let recipient = *vector::borrow(&recipients, 0);
+            let current_shares = pool_u64_iterable::shares(&distribution_pool, recipient);
+            let amount_to_distribute = pool_u64_iterable::redeem_shares(
```

Comment:
Will this guarantee the removal of the recipient from the delegation pool? In other words, will the remaining shares be reduced to exactly 0, even accounting for potential rounding errors in arithmetic?

Reply:
yes, this redeems all shares of a shareholder and that account is removed from the shareholders list too in `pool_u64` module. this logic is the same same as the one used in `staking_contract::distribute_internal`